### ISSUE TITLE
Minimum Postgresql version increased to 9.2

### ISSUE THREAD

**Issue by davidread** [ckan/ckan#2555]
Following proposal here: https://github.com/ckan/ideas-and-roadmap/issues/117
we think Postgres 9.2 is a good compromise, bringing in basic JSON column type, and it is supported by recent linux distributions.

- [ ] update source install instructions
- [ ] update package install instructions
- [ ] Travis - remove 8.4 test and make it test on 9.2 and 9.4?
- [ ] remove postgres 8.x code in datastore

**Comment by metaodi** [comment]
Is this change planned for CKAN 2.5? We currently have some installations of CentOS where PostgreSQL 9.2 is not available because of internal restrictions.

I agree with changing it, just wanted to know if there is a known timeframe for it, so we can plan accordingly.

**Comment by davidread** [comment]
Yes, planned for CKAN 2.5 in 3 months. We're aware that upgrading will
ruffle some feathers with bigger orgs, but 9.2 is almost 3 years old so
reasonably conservative.

Dave

**Comment by davidread** [comment]
Update: I've had all positive feedback so far, so I'm still planning to do this for CKAN 2.5

**Comment by davidread** [comment]
This change depends on us moving to Ubuntu 14.04 (which comes with Postgres 9.3, whereas Ubuntu 12.04 comes with 9.1). @amercader are you still gunning to get that (https://github.com/ckan/ckan/issues/1651) in for the CKAN 2.5 branch in early October?

And I guess we should have some upgrade instructions to cover upgrading Postgres, that covers Ubuntu 12.04 & 14.04, source & package installs. I've added that to the checkbox list above.

**Comment by davidread** [comment]
Update: I did much of the docs to show how to upgrade one's postgres database from 8.4:

https://github.com/ckan/ckan/compare/postgres-version

however I didn't get the postgis_restore.pl script to work. It's not something that is a priority for my work, so unassigning myself.

**Comment by NicolaiMogensen** [comment]
I get this error when trying to execute this command `sudo -u postgres psql --cluster 9.5/main -f backup_ckan.sql`

error: 
`The input is a PostgreSQL custom-format dump.
Use the pg_restore command-line client to restore this dump to a database.`
